test/test_codegen.py

```diff
@@ -29,16 +29,123 @@
+    sol.TPX = t, ct.one_atm, x
+    sol.equilibrate("UV")
+    t, rho, y = sol.TDY
+    p_ct = sol.P
+    # Compute pressure with pyrometheus and compare to Cantera
+    p_pm = ptk.get_pressure(rho, t, y)
+    assert abs(p_ct - p_pm) / p_ct < 1.0e-2
+
+
+def test_get_temperature():
+    """This function tests that pyrometheus-generated code
+    computes the right temperature for given internal energy
+    and mass fractions"""
+    # Create Cantera and pyrometheus objects
+    sol = ct.Solution("sanDiego.cti", "gas")
+    ptk = pyro.gen_python_code(sol)()
+    # Tolerance- chosen value keeps error in rate under 1%
+    tol = 1.0e-2
+    # Test temperatures
+    temp = np.linspace(500.0, 3000.0, 10)
+    # First test individual species
+    y = np.zeros(ptk.num_species)
+    for sp in range(ptk.num_species):
+        y[sp] = 1.0
+        for t in temp:
+            sol.TPY = t, ct.one_atm, y
+            e = sol.int_energy_mass
+            t_guess = 0.9 * t
+            t_pm = ptk.get_temperature(e, t_guess, y, True)
+            assert np.abs(t - t_pm) < tol
+        y[sp] = 0.0
+
+    # Now test a mixture with fully-populated composition
+    # All mass fractions set to the same value for now,
+    # though a more representative test would be ignition composition
+    y = np.ones(ptk.num_species) / ptk.num_species
+    for t in temp:
+        sol.TPY = t, ct.one_atm, y
+        e = sol.int_energy_mass
+        t_guess = 0.9 * t
+        t_pm = ptk.get_temperature(e, t_guess, y, True)
+        assert np.abs(t - t_pm) < tol
+
+
+def test_get_thermo_properties():
+    """This function tests that pyrometheus-generated code
+    computes thermodynamic properties c_p, s_R, h_RT, and k_eq
+    correctly by comparing against Cantera"""
+    # Create Cantera and pyrometheus objects
+    sol = ct.Solution("sanDiego.cti", "gas")
+    ptk = pyro.gen_python_code(sol)()
+    # Loop over temperatures
+    temp = np.linspace(500.0, 3000.0, 10)
+    for t in temp:
+
+        # Get properties from pyrometheus
+        cp_pm = ptk.get_species_specific_heats_R(t)
+        s_pm = ptk.get_species_entropies_R(t)
+        h_pm = ptk.get_species_enthalpies_RT(t)
+        keq_pm = 1.0 / np.exp(ptk.get_equilibrium_constants(t))
+        # Set state in cantera for comparison
+        sol.TP = t, ct.one_atm
+        keq_ct = sol.equilibrium_constants
+        # Compare properties
+        cp_err = (cp_pm - sol.standard_cp_R).max()
+        s_err = (s_pm - sol.standard_entropies_R).max()
+        h_err = (h_pm - sol.standard_enthalpies_RT).max()
+        keq_err = ((keq_pm - keq_ct) / keq_ct).max()
+        assert cp_err < 1.0e-14
+        assert s_err < 1.0e-14
+        assert h_err < 1.0e-14
+        assert keq_err < 1.0e-13
+
+    return
+
+
+# def test_sandiego():
+
+#     sol = ct.Solution("sanDiego.cti", "gas")
+#     print(pyro.gen_python_code(sol))
+#     #pyro_code = pyro.gen_python_code(sol)
+#     #ptk = pyro_code()
+#     #test_thermo(ptk, sol)
+
+#     return
```

Comment:
Just delete this. It can always be fished out of git.

Reply:
Will be reflected in next commit.